src/GildedItemFunctions.php

```diff
@@ -9,30 +9,26 @@ class GildedItemFunctions
 
     public function updateQuality(int $quality, ?int $sellIn): int
     {
-        if ($this->isQualityValueValid($quality)) {
-            if ($this->isQualityNotMaxed($quality)) {
-                $quality = $this->increaseQualityBy($quality, 1);
-                if ($sellIn < 11) {
-                    if ($this->isQualityNotMaxed($quality)) {
-                        $quality = ($sellIn < 6)
-                            ? $this->increaseQualityBy($quality, 2)
-                            : $this->increaseQualityBy($quality, 1);
-                    }
-                }
-            }
+        if (!$this->isQualityValueValid($quality)) {
+            return $quality;
+        }
+        if (!$this->isQualityValueValid($quality, true)) {
+            return $quality;
+        }
+        if (!($sellIn < 11)) {
+            return $quality;
         }
 
-        return $quality;
-    }
+        $quality = $this->increaseQualityBy($quality, 1);
 
-    private function isQualityValueValid(int $quality): bool
-    {
-        return $quality >= self::MIN_QUALITY && $quality <= self::MAX_QUALITY;
+        return ($sellIn < 6)
+            ? $this->increaseQualityBy($quality, 2)
+            : $this->increaseQualityBy($quality, 1);
     }
 
-    private function isQualityNotMaxed(int $quality): bool
+    private function isQualityValueValid(int $quality, bool $inRange = false): bool
     {
-        return $quality < self::MAX_QUALITY;
+        return ($inRange) ? $quality >= self::MIN_QUALITY && $quality <= self::MAX_QUALITY : $quality < self::MAX_QUALITY;
```

Comment:
1. Nadal nie wiem po co to jest sprawdzane 2x.
2. takie $inRange to jest zła praktyka, bo zachowanie metody jest w 100% inne od ustawienia tego parametru. To powinny być po prostu dwie różne metody. Nie musisz tego punktu poprawiać, bo jak poprawisz pkt 1 to te $inRange nie będzie potrzebne.